src/endpoint/s3/ops/s3_get_bucket.js

```diff
@@ -73,8 +83,21 @@ async function get_bucket(req) {
                 Size: obj.size,
                 Owner: (!list_type || req.query['fetch-owner']) && s3_utils.DEFAULT_S3_USER,
                 StorageClass: s3_utils.parse_storage_class(obj.storage_class),
+            };
+
+            if (restore_status_requested && obj.restore_status) {
+                const restore_status = obj.restore_status;
+                contents.RestoreStatus = {
+                    IsRestoreInProgress: restore_status.ongoing,
+                };
+
+                if (!restore_status.ongoing && restore_status.expiry_time) {
+                    contents.RestoreStatus.RestoreExpiryDate = new Date(restore_status.expiry_time).toUTCString();
+                }
```

Comment:
can you move this logic to its own function?
it can simply return undefined to have no RestoreStatus element.
this will just leak less details to the calling context - something like
```
RestoreStatus: get_object_restore_status(obj, restore_status_requested),
```

Reply:
A separate function in `s3_get_bucket`? Or something in `object_sdk` which forwards the request to `namespace_fs`?